js/ad_manager_vast.js

```diff
@@ -1875,6 +2127,625 @@ OO.Ads.manager(function(_, $) {
+        tracking: this._getTracking($mediaNode[0])
+      });
+
+      var impressions = this._getImpressions();
+      var tracking = this._getTracking(isLinear ? $node[0] : $node.parent()[0]);
+      var errorTracking = _cleanString(this.$_node.find('Error').text());
+      var videoClickTracking;
+      if (isLinear) {
+        videoClickTracking = {
+          clickTracking: _cleanString(this.$_node.find('ClickTracking').text()),
+          clickThrough: _cleanString(this.$_node.find('ClickThrough').text()),
+          customClick: _cleanString(this.$_node.find('CustomClick').text())
+        };
+      } else {
+        videoClickTracking = {
+          nonLinearClickThrough: _cleanString(this.$_node.find('NonLinearClickThrough').text())
+        }
+      }
+
+      var sequence = this.$_node.attr('sequence');
+      var adPodLength = this.$_node.parent().find('[sequence] Linear').length;
+      if (!supportsPoddedAds(version) || !_.isNumber(parseInt(sequence))) {
+        sequence = null;
+      }
+
+      var companionAds = [];
+      var companions = $companionsNode.find('Companion');
+      if (companions.length !== 0) {
+        companions.each(function(i, v){
+          companionAds.push(parseCompanionAd($(v)));
+        });
+      }
+      // this is for linear/nonlinear
+      var ad = {
+        mediaFiles: mediaFile,
+        tracking: tracking,
+        duration: isLinear ? this.$_node.find('Duration').text() : 0,
+        skipOffset: $node.attr('skipoffset') || null
+      };
+      _.extend(ad, videoClickTracking);
+
+      var data = {
+        adType: 'vpaid',
+        companion: companionAds,
+        error: errorTracking,
+        impression: impressions,
+        linear: ad,
+        nonLinear: ad,
+        title: _cleanString(this.$_node.find('AdTitle').text()),
+        tracking: tracking,
+        type: isLinear ? this.amc.ADTYPE.LINEAR_VIDEO : this.amc.ADTYPE.NONLINEAR_OVERLAY,
+        version: version,
+        videoClickTracking: videoClickTracking
+      };
+
+      var result = {
+        adPodIndex: parseInt(sequence) || 1,
+        sequence: sequence || null,
+        adPodLength: adPodLength ? adPodLength : 1,
+        data: data,
+        fallbackAd: null,
+        positionSeconds: adLoaded.time / 1000,
+        adParams: adParams,
+        streams: {'mp4': ''},
+        type: AD_TYPE.INLINE,
+        mediaFile: mediaFile,
+        version: version,
+        durationInMilliseconds: OO.timeStringToSeconds(ad.duration) * 1000
+      };
+
+      return result;
+    }, this);
+
+    /**
+     * Starts the click-to-linear ad
+     * This is only required for VPAID ads
+     * @private
+     * @method Vast#_beginAd
+     */
+    var _beginAd = _.bind(function() {
+      var ad = currentAd.vpaidAd,
+          adLinear = ad.getAdLinear();
+
+      initSkipAdOffset(currentAd);
+      //Since IMA handles its own UI, we want the video player to hide its UI elements
+      this.amc.hidePlayerUi();
```

Comment:
There is an extra call to hidePlayerUI up in the playAd function. Can you check if one of these can be removed without affecting anything? Thanks!